src/libs/actions/Session.js

```diff
@@ -112,7 +133,82 @@ function signOut() {
         .catch(error => Onyx.merge(ONYXKEYS.SESSION, {error: error.message}));
 }
 
+/**
+ * Checks the API to see if an account exists for the given login
+ *
+ * @param {String} login
+ */
+function hasAccount(login) {
+    API.GetAccountStatus({email: login})
+        .then((response) => {
+            if (response.jsonCode === 200) {
+                Onyx.merge(ONYXKEYS.CREDENTIALS, {
+                    login,
+                });
+                Onyx.merge(ONYXKEYS.ACCOUNT, {
+                    accountExists: response.accountExists,
+                    canAccessExpensifyCash: response.canAccessExpensifyCash,
+                });
+            }
+        });
+}
+
+/**
+ * Create a login or an account depending on if the user's account they are logging in
+ * with already had an account associated with it or not.
```

Comment:
Somewhat confused about which two scenarios we are handling here. Maybe this comment can be more specific and also touch on what the difference is between a `login` and an `account` in this context? Anyone familiar with our API will have some ideas about what is meant here. But contributors are likely to be confused.

Separately, I think it might also be clearer if there were separate methods for handling the specific cases. e.g. if the scenarios are either logging in with an account that exists or creating an account w/ a user provided login & password then the latter seems more like a `signUp` than a `signIn`.